ui/src/app/utils/navigation.tsx

```diff
@@ -58,6 +58,15 @@ export const navigateByUrl = (...args) => {
   return NavStore.navigateByUrl(...args);
 };
 
+// if modifier keys are pressed (like shift or cmd) use the href
+// if no keys are pressed, prevent default behavior and route using navigateByUrl
+export const navigateAndPreventDefaultIfNoKeysPressed = (e: React.MouseEvent, url: string) => {
```

Comment:
Optional: a slightly shorter / pithier name could be "navigateOrAllowClick"? I'm not sure if it still correctly reflects the intent of the method, but I found the name as-is to be a bit of a mouthful (to the point where it detracts from readability / comprehension)